### DISCUSSION TITLE
What is the difference between `rauc resign` and `rauc replace-signature` ?

### DISCUSSION THREAD

**Discussion by FederAndInk** [rauc/rauc#1366] [Q&A] [answered]
Hello, I am a bit confused about the differences between `rauc resign` and `rauc replace-signature`

Are they achieving a similar purpose? or is one recommended over the other?

Thanks :)

**Comment by jluebbe** [comment, marked as answer]
`rauc resign` generates the new signature just like `rauc bundle`, just for an existing bundle. `rauc replace-signature` expects the signature as a binary file and is for use with special cases like offline signing.

**Reply by FederAndInk** [reply]
so, if I understand correctly, if another process/tool is generating the signature I need to use replace-signature then

**Reply by jluebbe** [reply]
You'd use it together with `rauc extract-signature` to get original signature generated by RAUC and then generate your own with a custom tool/script.

**Reply by FederAndInk** [reply]
alright, thanks :)
